cpp/test/repartition_test.cpp

```diff
@@ -0,0 +1,259 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "common/test_header.hpp"
+#include "test_utils.hpp"
+#include <cylon/table.hpp>
+
+namespace cylon {
+namespace test {
+
+static void verify_test(std::vector<std::vector<std::string>>& expected, std::shared_ptr<Table>& output) {
+    std::stringstream ss;
+    output->PrintToOStream(ss);
+    std::string s;
+    int i = 0;
+    while(ss>>s) {
+        REQUIRE(s == expected[RANK][i++]);
+    }
+    REQUIRE(i == expected[RANK].size());
+}
```

Comment:
you can use `ARROW_EQUALS` macro here. To create expected arrow table, use `TableFromJSON` in arrow_test_utils header

Reply:
bump